docs/coding.md

```diff
@@ -0,0 +1,102 @@
+# Coding Conventions
+
+Please follow coding conventions and guidelines described in the following documents:
+
+* [Go proverbs](https://go-proverbs.github.io/) - highly recommended read
+* [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments)
+* [Effective Go](https://golang.org/doc/effective_go.html)
+* [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+Here's a list of some more specific conventions that are often followed in
+the code and will be pointed out in the review process:
+
+## General
+
+* Keep variable names short for variables that are local to the function.
+* Do not export a function or variable name outside the package until you
+  have an external consumer for it.
+
+### Imports
+
+We use the following convention for specifying imports:
+
+```
+<import standard library packages>
+
+<import ceph-csi-operator packages>
+
+<import third-party packages>
+```
+
+Example:
+
+```go
+import (
+ "os"
+ "path"
+ "strings"
+ "time"
+
+ "github.com/ceph/ceph-csi-operator/util"
+
+ "github.com/pborman/uuid"
+)
+```
+
+### Error Handling
+
+* Use variable name `err` to denote error variable during a function call.
+* localize error vars to the deepest scope possible, this allows to reuse the
+  err var name and have a different memory location for each. For example, do
+  not use `errWrite` or `errRead`.
+* Do not panic() for errors that can be bubbled up back to user. Use panic() in
+  higher function only for fatal errors which shouldn't occur.
+* Do not ignore errors using `_` variable unless you know what you're doing.
+* Error strings should not start with a capital letter.
+* If error requires passing of extra information, you can define a new type
+* Error types should end with `Error`. Reuse existing error types as much as
+  possible.
```

Comment:
Another example of how hard bracks make it really hard to read and review
![image](https://github.com/ceph/ceph-csi-operator/assets/14993359/ac3508b0-f7d1-410d-a64f-d8bf729d6fbf)

Asking to remove any hard breaks from these lines and others in this file

Reply:
Removed it, i dont see it with git https://github.com/ceph/ceph-csi-operator/blob/c7a537ba3a806b9ae65e67e4a0345be6d34992dc/docs/coding.md